polyfill/spatial-navigation-polyfill.js

```diff
@@ -332,25 +332,61 @@
     return false;
   }
 
+  /**
+   * Find the candidates within a spatial navigation container include delegable container.
+   * This function does not search inside delegable container or focusable container.
+   * In other words, this return candidates set is not included focusable elements inside delegable container or focusable container.
+   *
+   * @function getSpatialNavigationCandidates
+   * @param container {Node} - The spatial navigation container
+   * @param option {FocusableAreasOptions} - 'mode' attribute takes visible' or 'all' for searching the boundary of focusable elements.
+   *                                          Default value is 'visible'.
+   * @returns {sequence<Node>} candidate elements within the container
+   */
+  function getSpatialNavigationCandidates(container, option = {'mode': 'visible'}) {
+    let candidates = [];
+
+    if (container.childElementCount > 0) {
+      if (!container.parentElement) {
+        container = document.body;
+      }
+      const children = container.children;
+      for (const elem of children) {
+        if (isDelegableContainer(elem)) {
+          candidates.push(elem);
+        } else if(isFocusable(elem)) {
+          candidates.push(elem);
+
+          if(!isContainer(elem) && elem.childElementCount) {
+            candidates = candidates.concat(getSpatialNavigationCandidates(elem));
```

Comment:
I think it would need to filter candidates which are not inside the container.
For example, sometimes the focus moves unnaturally as the sample below:
- Test case: http://10.177.242.47/jihye/spatnav-testcases/polyfill_focusable_inside_focusable.html

Reply:
It's very complex issue. I added some code to filter candidates which are not inside the container visually in filteredCandidates(). but I couldn't sure it's right concept. It looks need more discussion.

Reply:
For now, I think this is good to resolve. Let's discuss more on this issue with https://github.com/w3c/csswg-drafts/issues/3386.